lib/VMwareWebService/broker_timeout.rb

```diff
@@ -0,0 +1,64 @@
+module Timeout
```

Comment:
Well, yes, I'm wrong to say timeout is broken.  It's clearly not protecting a critical section where the yielded block completes but the timeout thread raises before it is killed.  Clearly, there may be a bug elsewhere in the thread code that exacerbates the problem.